compiler/AST/AstToText.cpp

```diff
@@ -1372,3 +1389,36 @@ bool AstToText::isMtArg(CallExpr* expr, bool expectThis) const
   return retval;
 }
 
+void AstToText::appendEnumDecl(EnumType* et) {
+  mText += et->symbol->name;
+
+  appendEnumConstants(et);
+}
+
+void AstToText::appendEnumConstants(EnumType* et) {
+  int count = et->constants.length;
+  bool first = true;
+
+  if (count > 0) {
+    mText += " { ";
+
+    for_alist(constant, et->constants) {
+      if (DefExpr* de = toDefExpr(constant)) {
+        if (!first) {
+          mText += ", ";
+        } else {
+          first = false;
+        }
+        appendExpr(de, false);
+      } else {
+        // LYDIA 2015/04/06 Debugging support.  Means that the list of enum
+        // constants contained something that wasn't a DefExpr, which is
+        // unexpected
+        mText += " appendEnumConstants.00";
```

Comment:
I am supportive of starting to move towards swapping those “weirdo strings” out for more use of INT_ASSERT.

The markers were good for me in the earliest phases when I was focussed on trying to remove captureString()
and rely only on AstToText.  I could see that progress was being made and then search carefully for the various
markers in the output to see what I was missing.

I share Thomas’ view that the use of AstToText for chapel doc makes this approach less attractive.

Mike



On Apr 7, 2015, at 11:16 AM, Thomas Van Doren <notifications@github.com<mailto:notifications@github.com>> wrote:


In compiler/AST/AstToText.cpp<https://github.com/chapel-lang/chapel/pull/1808#discussion_r27904948>:

> +  if (count > 0) {
> +    mText += " { ";
> +
> +    for_alist(constant, et->constants) {
> +      if (DefExpr* de = toDefExpr(constant)) {
> +        if (!first) {
> +          mText += ", ";
> +        } else {
> +          first = false;
> +        }
> +        appendExpr(de, false);
> +      } else {
> +        // LYDIA 2015/04/06 Debugging support.  Means that the list of enum
> +        // constants contained something that wasn't a DefExpr, which is
> +        // unexpected
> +        mText += " appendEnumConstants.00";


To make the case a little stronger for INT_ASSERT()/INT_FATAL(), I think it would be easy to miss these extra bits of text in the chpldoc output (as an example). Using compiler errors will make the issues more obvious if/when they come up.

That said, it seems ok for that to be another effort to revise AstToText. Thanks!

—
Reply to this email directly or view it on GitHub<https://github.com/chapel-lang/chapel/pull/1808/files#r27904948>.